mm2src/lp_ordermatch/my_orders_storage.rs

```diff
@@ -65,12 +65,17 @@ pub async fn save_my_new_taker_order(ctx: MmArc, order: &TakerOrder) {
     }
 }
 
-pub async fn save_maker_order_on_update(ctx: MmArc, order: &MakerOrder) {
+pub async fn save_maker_order_on_update(ctx: MmArc, order: &mut MakerOrder, new_change: HistoricalOrder) {
     let storage = MyOrdersStorage::new(ctx);
+    if let Ok(old_order) = storage.load_active_maker_order(order.uuid).await {
+        order.changes_history = old_order.changes_history;
+        order.changes_history.get_or_insert(Vec::new()).push(new_change);
+    }
     storage
         .update_active_maker_order(order)
         .await
         .error_log_with_msg("!update_active_maker_order");
+    order.changes_history = None;
```

Comment:
It seems that it will cause the `order_status` to always return empty `changes_history`.

Reply:
Yeah, that seems to be the case for active orders. I can always get the `order_status` from the files as a solution, what do you think?

Reply:
I think it's fine. Please lock `maker_orders` just in case to avoid data races.